subsys/bluetooth/host/id.c

```diff
@@ -1612,14 +1612,15 @@ int bt_le_oob_get_local(uint8_t id, struct bt_le_oob *oob)
 			return -EINVAL;
 		}
 
+#if (CONFIG_BT_ID_MAX > 1)
 		if (IS_ENABLED(CONFIG_BT_OBSERVER) &&
 		    id != BT_ID_DEFAULT &&
```

Comment:
Can't we do something like this? 
```suggestion
		if (IS_ENABLED(CONFIG_BT_OBSERVER) &&
		    CONFIG_BT_ID_MAX > 1 &&
		    id != BT_ID_DEFAULT &&
```